packages/jest-jasmine2/src/PCancelable.ts

```diff
@@ -26,6 +26,7 @@ export default class PCancelable<T> extends Promise<T> {
       reject: (reason?: unknown) => void,
     ) => void,
   ) {
+    // @ts-expect-error
```

Comment:
@G-Rath I had to add these ignores now (from #10215). Any ideas on how to fix it without the ignore?

Reply:
What are the actual errors?

Reply:
hah, should have pushed without the ignore 🙈 

```
packages/jest-jasmine2/src/PCancelable.ts:29:22 - error TS2554: Expected 1 arguments, but got 0.

29     super(resolve => resolve());
                        ~~~~~~~~~

  node_modules/typescript/lib/lib.es2015.promise.d.ts:33:34
    33     new <T>(executor: (resolve: (value: T | PromiseLike<T>) => void, reject: (reason?: any) => void) => void): Promise<T>;
                                        ~~~~~~~~~~~~~~~~~~~~~~~~~
    An argument for 'value' was not provided.

packages/jest-jasmine2/src/PCancelable.ts:40:19 - error TS2345: Argument of type 'T | PromiseLike<T> | undefined' is not assignable to parameter of type 'T | PromiseLike<T>'.
  Type 'undefined' is not assignable to type 'T | PromiseLike<T>'.

40           resolve(val);
                     ~~~

[8:25:19 PM] Found 2 errors. Watching for file changes.
```

Reply:
probably related to the `new Promise<void>` changes I've had to make other places, but I wasn't able to make this one happy

Reply:
I think the answer is at least in part that we need to change `resolve` in the same way TS did in 4.1:

> resolve‘s Parameters Are No Longer Optional in Promises

So `resolve: (value?: T | PromiseLike<T>) => void,` needs to lose its `?`.

Reply:
The error on `super` seems correct to me: you're not passing in a value, which doesn't match the constructor of `Promise` 🤔 

```
super((resolve, reject) => executor(() => {}, resolve, reject));
```

That makes it happy, but I don't know if that'll have the correct behaviour.

Reply:
that leaves the first error

Reply:
Dunno 😀 We didn't `extend Promise` in the JS version before #10215, so I have no idea what it's supposed to do

Reply:
Yeah, we shared the prototype of Promise 😬 

```
Object.setPrototypeOf(PCancelable.prototype, Promise.prototype);
```

Reply:
Yeah, it was just a copy of https://github.com/sindresorhus/p-cancelable/blob/v2.0.0/index.js

Reply:
The answer might actually be to move the `this._promise` into the `super`, as that returns a promise.

```
super((resolve, reject) => {
      this._reject = reject;

      return executor(
        fn => {
          this._cancel = fn;
        },
        val => {
          this._pending = false;
          resolve(val);
        },
        err => {
          this._pending = false;
          reject(err);
        },
      );
    });
```

Reply:
Maybe replace `extends Promise<T>` with `implements PromiseLike<T>` - that doesn't give errors for me in this file but I've not checked if the rest of the project is typesafe.

Reply:
If it's easier we can just use some function instead of `new PCancelable`?

Reply:
we don't need `instanceof`, this is just internal to have a `cancel`

Reply:
That'd also probably work 🤷 

This is only used in `jest-jasmine2` internally, so shouldn't be too much work to refactor nor too breaking.

I'd see if it compiles & passes the tests with doing `implements PromiseLike<T>` first as that to me feels like the right change, and I think that should be fine as the only place this is actually used is in `queneRunner`.

That function returns an object with `cancel: () => void;` - while it binds the `cancel` to the `PCancelable` instance, there's no spec saying it's `this` is a Promise nor does it return a Promise for chaining.

So I think it should all be fine.

Reply:
`implements PromiseLike<T>` seems to work, yeah 🚀